src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1302,317 +1294,315 @@ private void IfDiagnostic(CompilationAnalysisContext context, StatementSyntax st
-                                    IfDiagnostic(context, statements[7], LocationIncorrectRule, statementIdentifierToken.Text, spanToken.Text);
+                                    IfDiagnostic(context, statements[3], EndSpanIncorrectRule, statementIdentifierToken.Text);
                                     return false;
                                 }
 
-                                if (statementCount > 8)
+                                if (!startSpanCorrect)
                                 {
-                                    SyntaxToken diagnosticToken = DiagnosticCreationCheck(ruleNames, locationToken, statements);
-                                    if (diagnosticToken == null || diagnosticToken.Text == "")
+                                    if (statementCount == 7)
                                     {
-                                        IfDiagnostic(context, statements[8], DiagnosticIncorrectRule, ruleNames[0], locationToken.Text);
+                                        var argument = ((statements[4] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[0];
+                                        ReportDiagnostic(context, StartSpanMissingRule, argument.GetLocation(), locationCorrect, keywordIdentifierToken.Text);
                                         return false;
                                     }
-
-                                    if (statementCount > 9)
-                                    {
-                                        bool reportCorrect = DiagnosticReportCheck(context, diagnosticToken, contextParameter, statements);
-                                        if (!reportCorrect)
-                                        {
-                                            IfDiagnostic(context, statements[9], DiagnosticReportIncorrectRule, contextParameter.Identifier.Text, diagnosticToken.Text);
-                                            return false;
-                                        }
-                                    }
                                     else
                                     {
-                                        ReportDiagnostic(context, DiagnosticReportMissingRule, statements[8].GetLocation(), contextParameter.Identifier.Text);
+                                        var argument = ((statements[5] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[0];
+                                        ReportDiagnostic(context, StartSpanMissingRule, argument.GetLocation(), locationCorrect, keywordIdentifierToken.Text);
                                         return false;
                                     }
                                 }
-                                else
+                                
+                                if (startSpanCorrect && endSpanCorrect)
```

Comment:
@jepetty Similarly, this should be an else on the other checks concerning startSpanCorrect and endSpanCorrect, as it would be the only remaining case